doc/Paso_a_paso_DIA_2.md

```diff
@@ -0,0 +1,185 @@
+
+>**Nota :pen:** En Python podemos escribir el nombre de una clase de diferentes formas. Hay algunas reglas que debemos cumplir: no debe contener espacios, los únicos caracteres permitidos son letras, número y el caracter `_` y solamente pueden comenzar con una letra o con el caracter `_`. No obstante, nosotros usamos la notación conocida como **PascalCasing**. Comenzamos el nombre simepre con letra mayúscula seguida de letras minúsculas, cada nueva palabra en el nombre comienza con una letra mayúscula. Por ejemplo: `Fruit`, `Snake`, `SnakeFruit`, `AwesomeClassExampleInPascalCase`. Recuerda siempre usar nombres cortos y descriptivos para las clases que definas.
+
+Lo siguiente es otro comentario aclarando qué contiene nuestra clase. Por último, añadimos el campo `position` que representa la posición de la fruta. Además aclaramos el tipo de datos de este campo. Analicemos esta línea:
+
+```python
+position: Tuple[int, int]
+```
+
+Lo primero es el nombre del campo. En este caso es `position`. Para los nombre de campos, variables y funciones usaremos otro tipo de notación: el **snake_casing**. Esta vez solamente usamos letras minúsculas y números, el caracter `_` lo usamos como sustituto del espacio. Por ejemplo: `position`, `previous_position`, `position_2`.
+
+Luego del nombre agregamos dos puntos `:` para indicar que vamos a definir el tipo de datos. En este caso el tipo es una tupla de dos enteros, uno para representar la posición horizontal y otro para la vertical. Esto lo definimos escribiendo `Tuple[int, int]`.
+
+>**Nota :pen:** En Python no es necesario definir los tipos de datos de los campos de una clase. Podemos solamente definir el nombre sin más. No obstante, es una buena práctica definirlos de esta manera pues de esta forma Python nos alertará cuando parezca que hacemos un mal uso de dicho campo. Por ejemplo, si más adelante en el código intentamos asignar una cadena de texto al campo `position` de una fruta, nos saltará una alerta diciendo que este campo está diseñado para contener una tupla de enteros. Esto nos protege de hacer malos usos de nuestros objetos y funciones por descuido.
+
+Ya tenemos nuestra clase `Fruit`, puedes copiar el código anterior en el archivo `src/model/fruit.py`. Ahora vamos a agregar también la clase `Snake`.
+
+Archivo **src/model/snake.py**
+```python
+"""Module with the representation of the Snake entity
+along with some util classes such as Direction.
+"""
+
+
+from enum import Enum
+from typing import List, Tuple
+
+
+class Direction(Enum):
+    """Enum class representing directions."""
+
+    RIGHT = 0
+    LEFT = 1
+    UP = 2
+    DOWN = 3
+
+
+class Snake:
+    """Representation of the Snake entity."""
+
+    def __init__(
+        self, position: Tuple[int, int], direction: Direction, size: int
+    ) -> None:
```

Comment:
I formatted the code using `black` because I'm used to doing that now. But maybe for beginners, we shouldn't format the code. What do you think? I don't know if it could bring up some confusion.

Reply:
Good point. I think it's better to remove these line breaks. It is too much to explain at the beginning.

Reply:
I also think that the `-> None` can be removed to be consistent with the other functions that return nothing